types.go

```diff
@@ -141,3 +145,332 @@ type Photo struct {
 	// htmlAttributions contains any required attributions.
 	HTMLAttributions []string `json:"html_attributions"`
 }
+
+// Component specifies a key for the parts of a structured address.
+// See https://developers.google.com/maps/documentation/geocoding/intro#ComponentFiltering for more detail.
+type Component string
+
+const (
+	// ComponentRoute matches long or short name of a route
+	ComponentRoute = Component("route")
+	// ComponentLocality matches against both locality and sublocality types
+	ComponentLocality = Component("locality")
+	// ComponentAdministrativeArea matches all the administrative_area levels
+	ComponentAdministrativeArea = Component("administrative_area")
+	// ComponentPostalCode matches postal_code and postal_code_prefix
+	ComponentPostalCode = Component("postal_code")
+	// ComponentCountry matches a country name or a two letter ISO 3166-1 country code
+	ComponentCountry = Component("country")
+)
+
+// Rankby specifies the order in which results are listed
+type Rankby string
+
+// Rankby options for Places Search
+const (
+	RankbyProminence = Rankby("prominence")
+	RankbyDistance   = Rankby("distance")
+)
+
+// PlaceType TODO(brettmorgan)
+type PlaceType string
+
+// Place Types for the Places API
+const (
```

Comment:
I have some code that has templates for generating both the Golang and Java versions of this enumeration. It is in a private repo at this point.